satellite/satellitedb/storagenodeaccounting.go

```diff
@@ -200,21 +200,41 @@ func (db *StoragenodeAccounting) QueryPaymentInfo(ctx context.Context, start tim
 func (db *StoragenodeAccounting) QueryStorageNodeUsage(ctx context.Context, nodeID storj.NodeID, start time.Time, end time.Time) (_ []accounting.StorageNodeUsage, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	query := `SELECT at_rest_total, start_time, 
-		LAG(at_rest_total) OVER win AS prev_at_rest, 
-		LAG(start_time) OVER win AS prev_start_time
-		FROM accounting_rollups
-		WHERE id IN (
-			SELECT MAX(id)
-			FROM accounting_rollups
-			WHERE node_id = ?
-			AND ? <= start_time AND start_time <= ?
-			GROUP BY start_time
-			ORDER BY start_time ASC
-		)
-		WINDOW win AS (ORDER BY start_time)`
+	lastRollup, err := db.db.Find_AccountingTimestamps_Value_By_Name(ctx, dbx.AccountingTimestamps_Name(accounting.LastRollup))
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+	if lastRollup == nil {
+		return nil, Error.New("unable to find last accounting rollup timestamp")
+	}
+
+	start, end = start.UTC(), end.UTC()
+
+	query := `SELECT SUM(at_rest_total), start_time
+				FROM (
+					SELECT r.at_rest_total, r.start_time
+					FROM (
+						SELECT MAX(id) as id
+						FROM accounting_rollups
+						WHERE node_id = ?
+						AND ? <= start_time AND start_time <= ?
+						GROUP BY start_time
+					) ids
+					INNER JOIN accounting_rollups r ON r.id = ids.id
+					UNION
+					SELECT SUM(data_total) as at_rest_total, DATE(interval_end_time) as start_time
+					FROM storagenode_storage_tallies
+					WHERE node_id = ?
+					AND ? < interval_end_time AND interval_end_time <= ?
+					GROUP BY start_time
+				) stamps 
+				GROUP BY start_time
+				ORDER BY start_time ASC`
```

Comment:
What's the performance of this query over a large database?

Reply:
@egonelbre so for DB with 365 * 10000 rollups and 366 * 10000 tallies:
- 0.396 seconds for SQlite
- 0.332 seconds for Postgres

Reply:
Add it to the pr description as well.